core/lib/contract_verifier/src/lib.rs

```diff
@@ -353,6 +360,73 @@ impl ContractVerifier {
         Ok((info, compiled_identifier))
     }
 
+    /// Checks whether the compiler versions in the request match those in the CBOR metadata (if available).
+    /// If they do not match, the versions in the request are updated to match the metadata.
+    async fn update_request_compiler_versions(
+        &self,
+        request: &mut VerificationRequest,
+        deployed_metadata: &Option<DetectedMetadata>,
+    ) -> Result<(), ContractVerifierError> {
+        if let Some(DetectedMetadata::Cbor {
```

Comment:
Nit: this function does nothing if `deployed_metadata` is `None`, so it's better to accept `&DetectedMetadata` here and have `if let Some(..)` on the caller side.

Reply:
Agreed, refactored!